docs/docs/concepts/overview.md

```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
 TODO
 
 ## Replay & Backups
-TODO
+A job might need to be re-run (backfill) due to business requirement changes or other various reasons. Optimus provides 
+an easy way to do this using Replay feature. Replay accepts which job and range of date to be updated, validating it, 
+and clearing the tasks in the scheduler, making the tasks available to be picked up by the scheduler and got re-run. 
+Replay will also figure the dependents of the job and re-run all the downstream jobs within the same project just after 
+the upstream tasks are finished.
+
+When validating, Optimus checks if replay with same job and date are currently running and also check if the task 
+scheduler instances are still running to avoid any duplication and confusion. Optimus provides several replay workers so 
+several replays can be run in parallel.
+
+After the replay worker cleared the task instances in the scheduler, the worker finished its job and let the scheduler do 
```

Comment:
People don't know what replay worker is. 
"the worker finished its job", what worker, what job?